### PR TITLE
Added support for file in checkout ajax.

### PR DESCRIPTION
**Pull request by IamYashKapoor**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Added support for file in checkout ajax.

### How to test the changes in this Pull Request:

1. If we add any <input type="file" name="file_data"> in checkout form using hooks.
2. After clicking on place order button, type file data is not being serialized in jQuery serialize method.
3. Now, after adding this code, we are able to get all data of the form.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Dev - Added support for file in checkout AJAX.

### CODE REVIEW

**Review by claudiosanches — APPROVED** [review]
Works fine, but doesn't support IE9, not sure if we suppose to keep this support: https://caniuse.com/#feat=mdn-api_formdata_formdata

Anyway this is a modern solution, I prefer it instead of jQuery `serialize()`.

Thanks for your help!

**Comment by yash-webkul** [comment]
Hello @claudiosanches 
Can you please let me know, when it will get merged and comes in the update?

**Comment by yash-webkul** [comment]
Hello @claudiosanches 
Any response would be appreciated. Thanks!!

**Comment by claudiosanches** [comment]
Hello @yash-webkul, I can't tell when we'll include this PR, we need to get at least one more review on this PR to make sure we aren't breaking anything.

**Review by claudiosanches — APPROVED** [review]

**Comment by rodrigoprimo** [comment]
Merging as we don't require two reviews anymore and @claudiosanches already approved it.

**Comment by claudiosanches** [comment]
@rodrigoprimo I believe that this PR has great potential to break backward compatibility and cause conflicts with third party plugins. That's why I said that this PR needs a second look.

**Comment by vedanshujain** [comment]
Apologies for not reviewing in time, I started but I was not able to actually use a file field in the checkout form even after spending quite a bit of time, after which I got distracted with another issue. I also recommending reverting and not rushing it so that we can take a better look.

**Comment by rodrigoprimo** [comment]
Just documenting that we chatted a bit about this PR, and @vedanshujain will review it tomorrow and decide whether to keep it merged or to revert it.

I misinterpreted Claudio's comment about a second review. I thought he mentioned that because back when the comment was posted, we required two reviews for every PR and not because he felt it was important for this specific PR to get another review.

**Comment by ObliviousHarmony** [comment]
After examining the potential impact of this PR I've decided to revert it for 4.8 so that it can be more extensively tested.

In the interim you can implement the functionality using the `checkout_place_order` handler, which will allow you to short-circuit WooCommerce's AJAX request and replace it with your own that uses `FormData`.

**Comment by ObliviousHarmony** [comment]
I've prepared another PR to introduce this change in https://github.com/woocommerce/woocommerce/pull/28296.

**Pull request merged**
